sdk/core/core-tracing/src/createSpan.ts

```diff
@@ -68,28 +68,27 @@ export function createSpanFunction(args: CreateSpanFunctionArgs) {
     if (span.isRecording() && args.namespace) {
       newSpanOptions = {
         ...tracingOptions.spanOptions,
-        parent: span.context(),
         attributes: {
           ...spanOptions.attributes,
           "az.namespace": args.namespace
         }
       };
     }
 
-    const newTracingOptions: OperationTracingOptions = {
+    const newTracingOptions: Required<OperationTracingOptions> = {
       ...tracingOptions,
-      spanOptions: newSpanOptions
-      // TODO: .context soon.
+      spanOptions: newSpanOptions,
+      context: setSpan(tracingOptions.context || otContext.active(), span)
     };
 
     const newOperationOptions = {
       ...operationOptions,
       tracingOptions: newTracingOptions
-    };
+    } as T & { tracingOptions: Required<OperationTracingOptions> };
```

Comment:
do we need the `as` or could we type the const directly?

Reply:
This is one of those conditions where, AFAIK, you're required to use `as`. If I just use this as the declared type the error is this:

```
Type '{ tracingOptions: Required<OperationTracingOptions>; }' is not assignable to type 'T & { tracingOptions: Required<OperationTracingOptions>; }'.
  Type '{ tracingOptions: Required<OperationTracingOptions>; }' is not assignable to type 'T'.
    '{ tracingOptions: Required<OperationTracingOptions>; }' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint '{ tracingOptions?: OperationTracingOptions | undefined; }'.ts(2322)
```

My understanding is that this is because `T` could potentially be a distinct sibling to the object I've created. So the only person that can truly know the right thing is me (and I have to do that via `as`).